test/unit/plugins/rest_spec.js

```diff
@@ -646,4 +647,97 @@ describe("The Rest plugin", function () {
 			});
 		});
 	});
+
+	describe("updating a created instance", function () {
+
+		var server    = new Hapi.Server();
+		var instances = new InstanceAdapter();
+		var result;
+		var instanceID;
+		var updatedInstance;
+
+		before(function () {
+
+			server.connection();
+			return server.registerAsync({
+				register : Rest,
+				options  : {
+					instances : instances
+				}
+			});
+		});
+
+		after(function () {
+			return server.stopAsync();
+		});
+
+		describe("setting the status of a created instance to failed", function () {
+
+			before(function () {
```

Comment:
I think it would more accurately reflect the real world flow here to use the `InstanceAdapter` only for setting up preconditions and to perform a `GET` request to fetch and update the instance. This should also simulate the `revision` behavior that we want.